azure/scope/cluster.go

```diff
@@ -711,33 +711,37 @@ func (s *ClusterScope) GetPrivateDNSZoneName() string {
 }
 
 // APIServerLBPoolName returns the API Server LB backend pool name.
-func (s *ClusterScope) APIServerLBPoolName(loadBalancerName string) string {
-	return azure.GenerateBackendAddressPoolName(loadBalancerName)
+func (s *ClusterScope) APIServerLBPoolName() string {
+	return s.APIServerLB().BackendPool.Name
```

Comment:
Is it possible that `s.APIServerLB()` to be `nil`? If so, can we run into _nil pointer dereference_ ?

Reply:
Also, do you think it is valid to use `azure.GenerateBackendAddressPoolName(loadBalancerName)` as a backup name if `s.APIServerLB().BackendPool.Name` is nil?

Reply:
> Is it possible that s.APIServerLB() to be nil? If so, can we run into nil pointer dereference ?

It cannot be nil. A function above is creating the pointer from a struct field.
// APIServerLB returns the cluster API Server load balancer.
func (s *ClusterScope) APIServerLB() *infrav1.LoadBalancerSpec {
	return &s.AzureCluster.Spec.NetworkSpec.APIServerLB
}

`azure.GenerateBackendAddressPoolName(loadBalancerName)` is a valid default name for the backendPool, but it is already being set by the defaulting webhook if it is empty.